elastic.go

```diff
@@ -396,36 +427,86 @@ func cmdNode(c *cli.Context, subCmd string) string {
 	default:
 		route = ""
 	}
-	return url + route
+	return route
 }
 
 func cmdQuery(c *cli.Context) string {
 	route := c.Args().First()
-	url := c.GlobalString("baseurl")
-	return url + route
+	return route
 }
 
 func cmdStats(c *cli.Context, subCmd string) string {
 	var route string
-	url := c.GlobalString("baseurl")
+
 	switch subCmd {
 	case "size":
-		route = "_stats/index,store"
+		route = "_stats/indexing,store"
 	default:
 		route = ""
 	}
-	return url + route
+	return route
 }
 
-func httpGet(route string, trace bool) (*http.Response, error) {
-	if trace {
-		fmt.Fprintf(os.Stderr, "GET: %s", route)
+func httpGetAll(route string, c *cli.Context) ([]*http.Response, error) {
+	if c.GlobalBool("trace") {
+		fmt.Fprintf(os.Stderr, "GET: %s\n", route)
+	}
+
+	var rs []*http.Response
+	var err error
+	urls := strings.Split(c.GlobalString("baseurl"), ",")
+
+	if len(urls) == 0 {
+		if c.GlobalBool("trace") {
+			fmt.Fprintf(os.Stderr, "No URLS found in %v\n", c.GlobalString("baseurl"))
+		}
+
+		return nil, errors.New("no urls")
+	}
+
+	for _, url := range urls {
+		reqURL := url + "/" + route
+		r, e := http.Get(reqURL)
+
+		if e != nil {
+			if c.GlobalBool("trace") {
+				fmt.Fprintf(os.Stderr, "Failed when making request to %s with error %s\n", reqURL, err.Error())
+			}
+
+			// Don't want to error out if we fail to get a response from one of the nodes
+			err = e
+			continue
+		}
+
+		rs = append(rs, r)
 	}
-	r, err := http.Get(route)
 
-	return r, err
+	return rs, err
 }
 
-func isTraceEnabled(c *cli.Context) bool {
-	return c.GlobalBool("trace")
+func httpGet(route string, c *cli.Context) (*http.Response, error) {
```

Comment:
Will try and get a single response from any of the nodes available. Used for commands like `elastic index list` where the response is the same from every node.